source/image_perspective.py

```diff
@@ -0,0 +1,312 @@
+
+    # Define transform
+    transform_points = get_coordinates_transform_points_(image=image, verbosity_level=verbosity_level)
+    whiteboard_dimensions = get_whiteboard_dimensions_(transform_points, accept_image_loss=False, verbosity_level=verbosity_level)
+
+    whiteboard_size = common.location.Pos(x=whiteboard_dimensions[0], y=whiteboard_dimensions[1])
+    crop_modifier = min(1, min(screen_size.x / whiteboard_size.x, screen_size.y / whiteboard_size.y)) # [0 - 1], modifier, to scale image so it fits on the screen
+
+    destination_corners = [[0,0],[whiteboard_size.x,0],[whiteboard_size.x,whiteboard_size.y],[0,whiteboard_size.y]]
+    homography = cv2.getPerspectiveTransform(numpy.float32(transform_points), numpy.float32(destination_corners))
+
+    output_img = cv2.warpPerspective(image, homography, (whiteboard_size.x,whiteboard_size.y), flags=cv2.INTER_LINEAR)
+    if (verbosity_level > 0):
+        end_time = datetime.datetime.now()
+        print(f'Time:\n\tStart:\t{start_time}\n\tEnd:\t{end_time}\n\tElapsed:\t{end_time - start_time}')
+    if (verbosity_level > 1): cv2.imshow(f'Final transform',cv2.resize(output_img, (int(whiteboard_size.x * crop_modifier),int(whiteboard_size.y * crop_modifier))))
+    return output_img
+
+
+
+
+
+
+
+
+
+if __name__ == "__main__":
+    print ('Running demo image_perspective\nThis demo is about how to find the corners of an transformed rectangle fully automatically.\nDue to its GIRTH it is advised to only do it per user request or as part of a first time setup.')
+
+    # open an image
+    absolute_path = os.path.join(os.getcwd(), 'files', 'image_perspective', 'whiteboard.jpg')
+    # absolute_path = os.path.join(os.getcwd(), 'files', 'image_perspective', 'paper.jpg')
+    # absolute_path = os.path.join(os.getcwd(), 'files', 'image_perspective', 'paper.png') BUG
+    print(f'Opening ({absolute_path}) as an image.')
+    img = cv2.imread(absolute_path, 1)
+    
+
+    # Using another useful function that make the given image fit inside the current screen
+    img_size = common.location.Pos(x=img.shape[1], y=img.shape[0])
+    cv2.imshow(f'Input of demo ( width:{img_size.x}, height:{img_size.y} )',resize_image_to_fit_screen(img))
+    
+
+    # Using the function
+    output_img = warp_perspective(image=img, verbosity_level=2) # Note that a verbosity_level shows intermediate steps
+    output_img2 = warp_perspective(image=img, verbosity_level=1)
+
+
+    # Using another usefull function that make the given image fit inside the current screen
+    output_img_size = common.location.Pos(x=output_img.shape[1], y=output_img.shape[0])
+    cv2.imshow(f'Output of demo ( width:{output_img_size.x}, height:{output_img_size.y} )',resize_image_to_fit_screen(output_img))
+    cv2.imshow(f'Output of demo 2 ( width:{output_img_size.x}, height:{output_img_size.y} )',resize_image_to_fit_screen(output_img2))
+    
+
+    # Wait so we can visually validate
+    cv2.waitKey(delay=300000) # 5 minutes max
```

Comment:
This is a really long wait time we can't really afford if we want to adjust the sounds in real time

Reply:
This is the code that only runs when the library `image_perspective.py` is run itself.  
That means that this code only is only there to show how the library works. The actual code should be inside the `main.py`, but since i have a rather insignificant library for now (We are not yet doing anything with whiteboards) this seamed weird to implement. 😕 

Does this address your comment @Maucee ?